src/nvim/api/buffer.c

```diff
@@ -898,12 +921,17 @@ Object nvim_buf_get_var(Buffer buffer, String name, Error *err)
 Integer nvim_buf_get_changedtick(Buffer buffer, Error *err)
   FUNC_API_SINCE(2)
 {
-  const buf_T *const buf = find_buffer_by_handle(buffer, err);
+  buf_T *buf = find_buffer_by_handle(buffer, err);
 
   if (!buf) {
     return -1;
   }
 
+  // load buffer first if it's not loaded
+  if (!buf_ensure_loaded(buf)) {
+    api_set_error(err, kErrorTypeException, "Failed to load buffer");
+    return -1;
+  }
   return buf_get_changedtick(buf);
```

Comment:
this doesn't need to load the buffer. it should just passively return the initial value (all that matters is that it is a monotonic counter).